lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -124,8 +121,11 @@ class QcFlagService {
                 throw new BadParameterError(`You cannot insert flag for data pass (id:${dataPassId}), run (runNumber:${runNumber}), detector (name:${dplDetector.name}) as there is no association between them`);
             }
 
-            if (fromTime && (fromTime < targetRun.timeTrgStart.getTime() || targetRun.timeTrgEnd.getTime() < fromTime)
-                || toTime && (toTime < targetRun.timeTrgStart.getTime() || targetRun.timeTrgEnd.getTime() < toTime)) {
+            const isFromTimeBeyondReange = fromTime
+                && (fromTime < targetRun.timeTrgStart.getTime() || targetRun.timeTrgEnd.getTime() < fromTime);
+            const isToTimeBeyondReange = toTime
+                && (toTime < targetRun.timeTrgStart.getTime() || targetRun.timeTrgEnd.getTime() < toTime);
+            if (isFromTimeBeyondReange || isToTimeBeyondReange) {
```

Comment:
from time is before to time, so you can skip two conditions: just check that fromTime is before end time and to time is after start time, and you'll be sure that you are in the right period

Reply:
no, because they can be null